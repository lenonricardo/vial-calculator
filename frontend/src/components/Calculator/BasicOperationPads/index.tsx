import OperationProps from 'interfaces/OperationProps';
import {
  Wrapper,
  PrimaryPad,
  PrimaryGradientPad,
  SecondaryGradientPad
} from './styles'

interface BasicOperationProps extends OperationProps {
  onEquals: () => void;
}

enum BasicOperationsEnum {
  DIVISION = '/',
  MULTIPLICATION = '*',
  SUBSTRACTION = '-',
  ADDITION = '+',
  EQUALS = '='
}

export default function OperationPads({
  onInput,
  onEquals
}: BasicOperationProps) {
  const handleOperationClick = (operation: string) => {
    onInput(operation)
  }

  const handleEquals = () => {
    onEquals()
  }

  return (
    <Wrapper>
      <PrimaryPad
        onClick={() => handleOperationClick(BasicOperationsEnum.DIVISION)}
      >
        {BasicOperationsEnum.DIVISION}
      </PrimaryPad>

      <PrimaryGradientPad
        onClick={() => handleOperationClick(BasicOperationsEnum.MULTIPLICATION)}
      >
        {BasicOperationsEnum.MULTIPLICATION}
      </PrimaryGradientPad>

      <SecondaryGradientPad
        onClick={() => handleOperationClick(BasicOperationsEnum.SUBSTRACTION)}
      >
        {BasicOperationsEnum.SUBSTRACTION}
      </SecondaryGradientPad>

      <SecondaryGradientPad
        onClick={() => handleOperationClick(BasicOperationsEnum.ADDITION)}
      >
        {BasicOperationsEnum.ADDITION}
      </SecondaryGradientPad>

      <SecondaryGradientPad onClick={() => handleEquals()}>
        {BasicOperationsEnum.EQUALS}
      </SecondaryGradientPad>
    </Wrapper>
  )
}
