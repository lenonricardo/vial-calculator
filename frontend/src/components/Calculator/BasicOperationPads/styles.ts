import styled from 'styled-components'
import { Pad } from '../NumberPads/styles'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`

export const PrimaryPad = styled(Pad)`
  && {
    background: var(--theme-secondary);
    color: #fff !important;

    &:hover {
      background: var(--theme-secondary);
    }
  }
`

export const PrimaryGradientPad = styled(Pad)`
  && {
    background: linear-gradient(
      180deg,
      var(--theme-secondary) 9%,
      var(--theme-primary) 100%
    );
    color: #fff !important;
  }
`

export const SecondaryGradientPad = styled(Pad)`
  && {
    background: var(--theme-primary);
    color: #fff !important;

    &:last-child {
      border-radius: 0 0 1.5rem 0;
    }

    &:hover {
      background: var(--theme-primary);
    }
  }
`
