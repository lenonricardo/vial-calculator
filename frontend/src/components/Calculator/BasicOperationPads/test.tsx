import { render } from '@testing-library/react'

import BasicOperationPads from '.'

describe('<BasicOperationPads />', () => {
  it('should render the component', () => {
    const mockOnInput = jest.fn()
    const mockOnEquals = jest.fn()

    const { container } = render(
      <BasicOperationPads onInput={mockOnInput} onEquals={mockOnEquals} />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
