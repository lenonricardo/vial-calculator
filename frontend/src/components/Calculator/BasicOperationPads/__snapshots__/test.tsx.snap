// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicOperationPads /> should render the component 1`] = `
.c1.c1 {
  width: var(--pad-width);
  height: var(--pad-height);
  color: var(--font-pads-color);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: var(--default-font-size);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 0;
}

.c1.c1:hover {
  background-color: #fff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2.c2 {
  background: var(--theme-secondary);
  color: #fff !important;
}

.c2.c2:hover {
  background: var(--theme-secondary);
}

.c3.c3 {
  background: linear-gradient( 180deg, var(--theme-secondary) 9%, var(--theme-primary) 100% );
  color: #fff !important;
}

.c4.c4 {
  background: var(--theme-primary);
  color: #fff !important;
}

.c4.c4:last-child {
  border-radius: 0 0 1.5rem 0;
}

.c4.c4:hover {
  background: var(--theme-primary);
}

<div
  class="c0"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium c1 c2 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    /
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium c1 c3 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    *
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium c1 c4 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    -
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium c1 c4 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    +
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium c1 c4 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    =
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
